admin/test/test_merge_pr.py

```diff
@@ -0,0 +1,212 @@
+        self.assertEqual(status, merge_pr.final_status([status]))
+
+    @given(commit_statuses, commit_statuses)
+    def test_final_status_many(self, status1, status2):
+        target = status1
+        if status2['updated_at'] > status1['updated_at']:
+            target = status2
+        self.assertEqual(target, merge_pr.final_status([status2, status1]))
+
+    @given(commit_statuses)
+    def test_not_success(self, status):
+        assume(status['state'] != u"success")
+        self.assertEqual(True, merge_pr.not_success(status))
+
+    @given(commit_statuses)
+    def test_not_success_success(self, status):
+        status['state'] = u"success"
+        self.assertEqual(False, merge_pr.not_success(status))
+
+    @given(commit_statuses, jenkins_results)
+    def test_format_status(self, status, jenkins):
+        formatted = merge_pr.format_status((status, jenkins))
+        self.assertIsInstance(formatted, unicode)
+        self.assertIn(status['context'], formatted)
+        self.assertIn(status['target_url'], formatted)
+
+
+# These strategies are far from complete coverage of the possible
+# Jenkins API responses.
+
+jenkins_builds = fixed_dictionaries(dict(
+    result=sampled_from([None, 'FAILURE', 'ABORTED', 'SUCCESS'])))
+"""Strategy for generating records of individual builds of a Jenkins job."""
+
+
+jenkins_build_result_with_builds = fixed_dictionaries(dict(
+    inQueue=booleans(),
+    builds=lists(jenkins_builds),
+    property=dictionaries(text(), text())))
+"""Strategy for generating Jenkins API information for a job with builds."""
+
+jenkins_build_result_with_no_builds = fixed_dictionaries(dict(
+    inQueue=booleans()))
+"""Strategy for generating Jenkins API information for a job with no builds."""
+
+jenkins_build_results = one_of(
+    jenkins_build_result_with_builds,
+    jenkins_build_result_with_no_builds,
+    just(pmap()))
+"""Strategy for generating Jenkins API information for a job."""
+
+
+class JenkinsResultsTests(TestCase):
+    """
+    Tests for interpretation of build results from Jenkins.
+    """
+
+    @given(jenkins_build_results)
+    def test_result_types(self, info):
+        result, params = merge_pr.jenkins_info_from_response(info)
+        self.assertIn(result, list(merge_pr.JenkinsResults.iterconstants()))
+        if params is not None:
+            self.assertIsInstance(params, dict)
+
+    @given(jenkins_build_results)
+    def test_in_queue(self, info):
+        info = dict(info)
+        info['inQueue'] = True
+        result, params = merge_pr.jenkins_info_from_response(info)
+        self.assertEqual(merge_pr.JenkinsResults.RUNNING, result)
+        self.assertEqual({}, params)
+
+    @given(jenkins_build_results)
+    def test_builds_not_present(self, info):
+        assume(not info.get('inQueue', False))
+        info = dict(info)
+        if 'builds' in info:
+            del info['builds']
+        result, params = merge_pr.jenkins_info_from_response(info)
```

Comment:
:question: This test looks really weird, The assume will through out ~half of the examples which isn't great, and then after mutations I think you are left with something that looks like:
```
one_of(
  just(pmap()),
  just(pmap(inQueue=False)),
  fixed_dictionaries(dict(inQueue=just(False), property=dictionaries(text(), text())))
```

Hypothesis will still try to come up with a bunch of different `builds=lists(jenkins_builds)`.

I guess this is starting to get borderline with me as far as if this really gets that much benefit from using hypothesis, especially when combined with the assume's and the mutations causing hypothesis to generate a bunch of examples that won't even expand test coverage.